daml/Marketplace/Listing.daml

```diff
@@ -57,30 +61,45 @@ template Service
         do
           create CreateListingRequest with status = Active; ..
 
-      nonconsuming RequestDeleteListing : ContractId DeleteListingRequest
+      nonconsuming RequestDisableListing : ContractId DisableListingRequest
         with
           listingId : Text
         do
-          create DeleteListingRequest with ..
+          create DisableListingRequest with ..
 
     controller provider can
       nonconsuming CreateListing : ContractId Listing
         with
           createListingRequestCid : ContractId CreateListingRequest
+          providerId : Text
         do
           CreateListingRequest{..} <- fetch createListingRequestCid
           archive createListingRequestCid
           create Listing with ..
 
-      nonconsuming DeleteListing : ()
+      nonconsuming ListingFailure : ContractId CreateListingRequest
```

Comment:
as above; maybe return a different contract `CreateListingFailure`

Reply:
This can easily be done but be aware that it may make the UI part more complex... We should decide if this is a design pattern we want to follow in the code base so the code is consistent everywhere.